### PR TITLE
BRS-300: Arrange layout of metrics page, add resuable components

### PR DESCRIPTION
**Pull request by wilwong89**
### Jira Ticket:
BRS-300

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/BRS-300

### Description:
Took Cameron's reusable metric components and metrics filter and arranged the layout of the metrics page.

### CODE REVIEW

File: src/app/metrics/site-metrics/site-metrics.component.scss
Comment (cameronpettit) [inline src/app/metrics/site-metrics/site-metrics.component.scss:5]: I have a hunch most of these scss classes can be moved to inline bootstrap classes, or at least to their child component .scss file for consistency - typically we only define scss classes as a last resort, for things we cannot control with bootstrap, like specific colours and specific width/heights. It keeps styling consistent if we ever need to change things site-wide.

File: src/app/metrics/site-metrics/site-metrics.component.html
Comment (cameronpettit) [inline src/app/metrics/site-metrics/site-metrics.component.html:18]: `<app-basic-metric>` was created to display/animated the numerical metric values, we should be passing `passTotal` into this component within each `<app-metric-card>`

File: src/app/shared/components/ds-forms/base-input/base-input.component.ts
Comment (cameronpettit) [inline src/app/shared/components/ds-forms/base-input/base-input.component.ts:35]: `BaseInputComponent` needs to implement `OnInit`

File: src/app/shared/components/ds-forms/shortdate-picker/shortdate-picker.component.ts
Comment (cameronpettit) [inline src/app/shared/components/ds-forms/shortdate-picker/shortdate-picker.component.ts:55]: rm

**Review by cameronpettit — CHANGES_REQUESTED** [review]
Couple of thoughts:
- I know the ticket says to display the metrics, but I am not sure how comfortable I would be if this were to go to `prod` with dummy data and not true API calls. Personally I think we should be hiding these components until the backend is ready to support them, but ultimately that is up to Jess. I would want a nod from her before I give my approval.
- Please include some tests so we can keep up our coverage (I apologize I didnt get around to writing them while I was working on this ticket).
-  Based on our backlog on March 13, I recall the team discussing that the chart download/export functionality might get moved to within the chart components themselves. Did this get resolved?

Some of my nits are on code I wrote.

File: src/app/metrics/site-metrics/site-metrics.component.scss
Comment (wilwong89) [reply src/app/metrics/site-metrics/site-metrics.component.scss:5]: Didn't know that it should be a last resort. I usually use SCSS if there will be multiple components reusing the same CSS settings so that adjusting it once adjusts it everywhere. I've followed your advice and moved into bootstrap classes.

File: src/app/metrics/metrics.component.ts
Comment (wilwong89) [reply src/app/metrics/metrics.component.ts:11]: Not sure what needs to be done here.

File: src/app/metrics/site-metrics/site-metrics.component.html
Comment (wilwong89) [reply src/app/metrics/site-metrics/site-metrics.component.html:18]: Properly changed things to wrap `<app-basic-metric>` with `<app-metric-cards>`

File: src/app/shared/components/ds-forms/base-input/base-input.component.ts
Comment (wilwong89) [reply src/app/shared/components/ds-forms/base-input/base-input.component.ts:35]: Implemented

**Comment by wilwong89** [comment]
- I think we can display the doughnut chart data and basic metrics for all parks for now until the integration work is done. It pretty much displays the same data as the prior metric chart before so it preserves prior functionality as placeholder for later work.
- I'm going to need some assistance and time to write tests for Angular components. Tried to take a crack at it and it seems it would take me too much time without some guidance.
- As per the backlog, the per chart report is pushed to a later ticket. As per the standup this morning, re-adding the existing button for report downloads which keeps prior functionality.

File: src/app/metrics/metrics.component.ts
Comment (wilwong89) [reply src/app/metrics/metrics.component.ts:11]: Removed unnecessary ngOnInit and constructor.

File: src/app/metrics/site-metrics/site-metrics.component.ts
Comment (marklise) [inline src/app/metrics/site-metrics/site-metrics.component.ts:49]: I believe these are better off initialized to `undefined` instead of `0` as it will take advantage of the `|| '-'` mode of the basic-metric.component.html:
```
<h1 class="mb-0">{{ (_value | number : "1.0-0" : "en-US") || "-" }}</h1>
```

**Comment by cameronpettit** [comment]
> * I think we can display the doughnut chart data and basic metrics for all parks for now until the integration work is done. It pretty much displays the same data as the prior metric chart before so it preserves prior functionality as placeholder for later work.
> * I'm going to need some assistance and time to write tests for Angular components. Tried to take a crack at it and it seems it would take me too much time without some guidance.
> * As per the backlog, the per chart report is pushed to a later ticket. As per the standup this morning, re-adding the existing button for report downloads which keeps prior functionality.

Sure, your changes look good to me and your first point seems fair - so long as Mark's nits are implemented so the page layout is consistent when data is loading/loaded/not found. Last point also sounds good to me.

Unit tests are part of our definition of done, but pursuant of moving this ticket along, we could create a new ticket to capture the unit test aspect and assign it to you. That flexibility will unblock the metrics stuff reliant on this ticket while you draw up the UTs - but we should get approval from Jess. If we get that, then I'll approve this.

**Review by marklise — APPROVED** [review]

**Pull request merged**
